pkg/apis/kubeone/config/config.go

```diff
@@ -28,36 +28,11 @@ import (
 	kubeonescheme "github.com/kubermatic/kubeone/pkg/apis/kubeone/scheme"
 	kubeonev1alpha1 "github.com/kubermatic/kubeone/pkg/apis/kubeone/v1alpha1"
 	"github.com/kubermatic/kubeone/pkg/apis/kubeone/validation"
-	"github.com/kubermatic/kubeone/pkg/credentials"
 	"github.com/kubermatic/kubeone/pkg/terraform"
 
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
-// SetKubeOneClusterDynamicDefaults sets the dynamic defaults for a given KubeOneCluster object
-func SetKubeOneClusterDynamicDefaults(cfg *kubeoneapi.KubeOneCluster) error {
```

Comment:
* We will not store credentials in the API anymore to increase security and make it easier to work with credentials and the API
* There will be RuntimeValidation that checks are credentials present
* This might require promoting the API to v1alpha2